lighthouse-core/gather/gatherers/image-elements.js

```diff
@@ -162,6 +185,53 @@ function determineNaturalSize(url) {
   });
 }
 
+/**
+ * @param {LH.Crdp.CSS.CSSStyle} [style]
+ * @param {string} property
+ * @return {string | undefined}
+ */
+function findSizeDeclaration(style, property) {
+  if (!style) return;
+  const definedProp = style.cssProperties.find(({name}) => name === property);
+  if (!definedProp) return;
+  return definedProp.value;
+}
+
+/**
+ * Finds the most specific directly matched CSS font-size rule from the list.
+ *
+ * @param {Array<LH.Crdp.CSS.RuleMatch>} [matchedCSSRules]
+ * @param {string} property
+ * @returns {string | undefined}
+ */
+function findMostSpecificCSSRule(matchedCSSRules, property) {
+  /** @type {function(LH.Crdp.CSS.CSSStyle): string | undefined} */
+  const isDeclarationofInterest = (declaration) => findSizeDeclaration(declaration, property);
+  const rule = FontSize.findMostSpecificMatchedCSSRule(matchedCSSRules, isDeclarationofInterest);
+  if (!rule) return;
+  // @ts-expect-error style is guaranteed to exist if a rule exists
+  return findSizeDeclaration(rule.style, property);
+}
+
+/**
+ * @param {LH.Crdp.CSS.GetMatchedStylesForNodeResponse} matched CSS rules}
+ * @param {string} property
+ * @returns {string | undefined}
+ */
+function getEffectiveSizingRule({attributesStyle, inlineStyle, matchedCSSRules}, property) {
+  // CSS sizing can't be inherited.
+  // We only need to check inline & matched styles.
+  // Inline styles have highest priority.
+  const inlineRule = findSizeDeclaration(inlineStyle, property);
+  if (inlineRule) return inlineRule;
+
+  const attributeRule = findSizeDeclaration(attributesStyle, property);
+  if (attributeRule) return attributeRule;
```

Comment:
nit: here and elsewhere, after every `return` statement that is an early exit, can you make sure there is an empty newline directly after? helps for readability imo

Reply:
yes, sounds good! I'll do that